Update for Qt6

jmlich opened the pull request:
I am not sure if there is reasonable way to make it compatible with both Qt5 and Qt6.
Let me know if you want to include some of my changed, I will prepare separate pull request.

jmlich committed:
use lupdate and lrelease from QT_INSTALL_BINS

jmlich committed:
Changes needed for qt6

Thaodan reviewed (commented):
Are you sure that these changes are all for Qt 6? In most cases they apply to later Qt5 versions too.

I'm not sure about the QML changes.

jmlich commented:
Yes, big part can be applied also with older Qt. Testing down to Qt 5.6 is quite difficult for me as I need to install another virtual machine. Actually, I would like to do more tests with Qt6.

Thaodan commented:
You can check the Qt docs to which those changes apply to. Most changes that are breaking in Qt 6 where deprecated in later Qt 5 versions.

Thaodan commented:
I can test the changes on Qt 5.6.

rainemak commented on qml/MenuFingerterm.qml:
```diff
@@ -18,7 +18,7 @@
 
 import QtQuick 2.0
-import QtQuick.XmlListModel 2.0
+import QtQml.XmlListModel
```
This won't work with Qt5.x as version number is mandatory with it. One simple approach could go a follows

- Rename / copy qml/MenuFingerterm.qml to  qml/MenuFingerterm5.qml and qml/MenuFingerterm6.qml
- Have Qt6 changes in qml/MenuFingerterm6.qml and Qt5 in qml/MenuFingerterm5.qml
- Expose Qt major version to the root context or have it as a singleton
- On qml/Main.qml use Loader to correct menu

Something like this to Main.qml to replace MenuFingerterm
```
Loader {
    id: menu

    readonly property showing: item && item.showing

    anchors.fill: parent
    source: QtMajorVersion === 5 ? Qt.resolvedUrl("MenuFingerterm5.qml") : Qt.resolvedUrl("MenuFingerterm6.qml")
}
```

Point is that neither of these menu component won't get compiled when Main.qml is compiled.

pvuorela replied:
I wouldn't maybe even copy the whole MenuFingerterm.qml. It should be enough putting the ~10 line  XmlListModel snippet into qml version specific model files and load such on MenuFingerTerm based on qt version.

rainemak reviewed (changes requested):
[qml/MenuFingerterm.qml](https://github.com/sailfishos/fingerterm/pull/6/files#diff-2e95ae4c9898f3bfb2469df73bf14c2e4f127589a6aef94b94d94e4dd823982b) should be separated per major Qt version. That'd be relatively simple way to cope major version changes.

pvuorela commented on textrender.cpp:
```diff
@@ -76,7 +76,11 @@ TextRender::TextRender(QQuickItem *parent) :
+#if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
     iFont.setStyleHint(QFont::Monospace, QFont::StyleStrategy(QFont::PreferDefault | QFont::ForceIntegerMetrics));
+#else
+    iFont.setStyleHint(QFont::Monospace, QFont::StyleStrategy(QFont::PreferDefault));
```
Prefer default would be the default so does this do anything? For second thing is even the earlier ForceIntegerMetrics doing anything since there is QFontMetrics used for sizes as suggested by the deprecation notice?

jmlich committed:
Qt6 compatibility

jmlich committed:
fix linker errors

g++ -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -Wl,-rpath,/usr/lib -Wl,-rpath-link,/usr/lib -o fingerterm  main.o terminal.o textrender.o ptyiface.o util.o keyloader.o qrc_resources.o moc_ptyiface.o moc_terminal.o moc_textrender.o moc_util.o moc_keyloader.o   -lutil /usr/lib/libQt6Quick.so /usr/lib/libQt6OpenGL.so /usr/lib/libQt6Gui.so /usr/lib/libGLX.so /usr/lib/libOpenGL.so /usr/lib/libQt6QmlMeta.so /usr/lib/libQt6QmlModels.so /usr/lib/libQt6QmlWorkerScript.so /usr/lib/libQt6Qml.so /usr/lib/libQt6Network.so /usr/lib/libQt6Core5Compat.so /usr/lib/libQt6Core.so -lpthread -lGLX -lOpenGL
/usr/bin/ld: /tmp/ccnboAap.ltrans0.ltrans.o: in function `Terminal::keyPress(int, int, QString const&)':
<artificial>:(.text+0x24f2): undefined reference to `Terminal::ch_ESC'
/usr/bin/ld: <artificial>:(.text+0x256b): undefined reference to `Terminal::ch_ESC'
/usr/bin/ld: <artificial>:(.text+0x25f3): undefined reference to `Terminal::ch_ESC'
/usr/bin/ld: <artificial>:(.text+0x284f): undefined reference to `Terminal::ch_ESC'
